lib/update/cluster/phases/init.go

```diff
@@ -211,6 +214,40 @@ func (p *updatePhaseInit) Execute(context.Context) error {
 	return nil
 }
 
+// initTeleportAuthToken creates a provisioning token that teleport nodes will
+// use to authenticate with the auth server if it doesn't exist yet.
+func (p *updatePhaseInit) initTeleportAuthToken() error {
+	cluster, err := p.Backend.GetLocalSite(defaults.SystemAccountID)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	allTokens, err := p.Backend.GetSiteProvisioningTokens(cluster.Domain)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	for _, t := range allTokens {
+		if t.IsTeleport() && t.IsPersistent() {
```

Comment:
Can a teleport token be not persistent?

Reply:
Not as implemented currently, but I'd like to avoid the situation if somebody (even me) uses teleport expiring tokens for something in future and we'll end up in the same situation as we did last week.